src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets

```diff
@@ -40,14 +40,20 @@ Copyright (c) .NET Foundation. All rights reserved.
     <IsInnerBuild Condition="'$(TargetFramework)' != '' AND '$(TargetFrameworks)' != ''">true</IsInnerBuild>
     <SymbolPackageFormat Condition="'$(SymbolPackageFormat)' == ''">symbols.nupkg</SymbolPackageFormat>
     <AddPriFileDependsOn Condition="'$(MicrosoftPortableCurrentVersionPropsHasBeenImported)' == 'true'">DeterminePortableBuildCapabilities</AddPriFileDependsOn>
-    <NuspecOutputPath Condition="'$(NuspecOutputPath)' == ''">$(BaseIntermediateOutputPath)$(Configuration)\</NuspecOutputPath>
     <WarnOnPackingNonPackableProject Condition="'$(WarnOnPackingNonPackableProject)' == ''">false</WarnOnPackingNonPackableProject>
     <ImportNuGetBuildTasksPackTargetsFromSdk Condition="'$(ImportNuGetBuildTasksPackTargetsFromSdk)' == ''">false</ImportNuGetBuildTasksPackTargetsFromSdk>
     <AllowedOutputExtensionsInPackageBuildOutputFolder>.dll; .exe; .winmd; .json; .pri; .xml; $(AllowedOutputExtensionsInPackageBuildOutputFolder)</AllowedOutputExtensionsInPackageBuildOutputFolder>
     <AllowedOutputExtensionsInSymbolsPackageBuildOutputFolder Condition="'$(SymbolPackageFormat)' != 'snupkg'">.pdb; .mdb; $(AllowedOutputExtensionsInPackageBuildOutputFolder); $(AllowedOutputExtensionsInSymbolsPackageBuildOutputFolder)</AllowedOutputExtensionsInSymbolsPackageBuildOutputFolder>
     <AllowedOutputExtensionsInSymbolsPackageBuildOutputFolder Condition="'$(SymbolPackageFormat)' == 'snupkg'">.pdb</AllowedOutputExtensionsInSymbolsPackageBuildOutputFolder>
     <SuppressDependenciesWhenPacking Condition="'$(SuppressDependenciesWhenPacking)' == ''">false</SuppressDependenciesWhenPacking>
   </PropertyGroup>
+
+  <PropertyGroup>
+    <RestoreOutputPath Condition="'$(RestoreOutputPath)' == ''">$(MSBuildProjectExtensionsPath)</RestoreOutputPath>
```

Comment:
Why do you need this set globally?

Reply:
The NuGet.targets already set the RestoreOutputPath. 
They can't really work without one another so as such I'd just revert this part of the change.

Reply:
> The NuGet targets already set the `RestoreOutputPath`

in the Target here...

https://github.com/NuGet/NuGet.Client/blob/19aec47a81889a9bed937c404d6e11b0701bf086/src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets#L643-L654

Pack targets doesn't depend on this target to get `RestoreOutputPath`.

Reply:
I see. 

Pack can just depend on MSBuildProjectExtensionsPath then. No need to reuse a property that's not consistently used.

Reply:
I will remove it from global declaration.

> Pack can just depend on MSBuildProjectExtensionsPath then

Can I lock the property on both Pack and Restore targets?

Reply:
> Can I lock the property on both Pack and Restore targets?

Can you clarify this?

Reply:
Remove ` Condition=" '$(RestoreOutputPath)' == '' " ` and set `TreatAsLocalProperty`.

Reply:
There are only two instances on SDK where the property is used. I can add changes to the existing PR if it's okay with you.

Reply:
Remove ` Condition=" '$(RestoreOutputPath)' == '' " ` and set `TreatAsLocalProperty`.

There are only two instances (`ComposeStore.targets` and `CrossGen.targets`) on SDK where the property is used.

Can you confirm if they have a specific reason for this?

If they have a reason, then we can't make it a local property.

Reply:
I wouldn't go beyond ensuring that there's a sensible default equivalent to what the SDK was setting.

Reply:
Then I'll go ahead with the changes. Let us test first, then I'll revert any undesired ones.